spec/rubocop/cop/rspec/empty_hook_spec.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::RSpec::EmptyHook do
+  subject(:cop) { described_class.new(config) }
+
+  let(:config) { RuboCop::Config.new }
+
+  shared_examples 'empty hook' do |hook_type|
+    it 'registers offense with empty block' do
+      expect_offense(<<~RUBY)
+        #{hook_type} {}
+        #{'^' * hook_type.length} Empty hook detected.
+      RUBY
+    end
+
+    it 'registers offense with empty single-line do-end' do
+      expect_offense(<<~RUBY)
+        #{hook_type} do; end
+        #{'^' * hook_type.length} Empty hook detected.
+      RUBY
+    end
+
+    it 'registers offense with empty multi-line do-end' do
+      expect_offense(<<~RUBY)
+        #{hook_type} do
+        #{'^' * hook_type.length} Empty hook detected.
+        end
+      RUBY
+    end
```

Comment:
I understand that you are trying to cover all the syntactic cases here, but there's no difference is AST that RuboCop operates on between an empty, or single-line, or multi-line block, so it's not really necessary to have separate examples for those two.